src/coprocessor/endpoint.rs

```diff
@@ -810,6 +801,41 @@ impl<E: Engine> Endpoint<E> {
     }
 }
 
+// check_request_key_range return error iif the request key range cross two or
+// more buckets.
+fn check_request_key_range(
+    req_ctx: &ReqContext,
+    buckets: Option<Arc<pd_client::BucketMeta>>,
+) -> std::result::Result<(), Error> {
+    if buckets.is_none() {
+        return Ok(());
+    }
+    let request_bucket_version = req_ctx.context.buckets_version;
+    let latest_buckets = buckets.unwrap();
+    if request_bucket_version == 0 || request_bucket_version >= latest_buckets.version {
+        return Ok(());
+    }
+
+    // check the request key range cross two or more buckets.
+    let start_key = txn_types::Key::from_raw(&req_ctx.lower_bound);
+    let end_key = txn_types::Key::from_raw(&req_ctx.upper_bound);
+    let first_index = find_bucket_index(start_key.as_encoded(), &latest_buckets.keys);
+    let last_index = find_bucket_index(end_key.as_encoded(), &latest_buckets.keys);
+    let mut cross_buckets = true;
+    if let (Some(first_index), Some(last_index)) = (first_index, last_index) {
+        cross_buckets = last_index - first_index >= 2;
```

Comment:
better to define a constant value for easy modification later if needed.

Reply:
Instead of use bucket count as threshold, it might be more accurate to use estimated size of the buckets [first_index, last_index] by using BucketMeta's sizes.  For example, generally 256MB should be quite safe, but 1GB could lead to some timeout.

Reply:
good suggestion , I will change this checker .